### PR TITLE
pyrosimple: 2.7.0 -> 2.8.0

### PR DESCRIPTION
**Pull request by vamega**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by ne9z** [comment]
Seems that the build was not successful and more debugging is needed.  Also, you can add yourself as maintainer.

**Comment by vamega** [comment]
Sorry for the broken PR. Had used the updater-script and pushed it thinking it would work.
I've got a revision that looks like it's working, but will wait for tests to pass locally before pushing that.

Will add myself as a maintainer here.

**Comment by ne9z** [comment]
Looks good.  However "with" statements are considered ambiguous by some,
and I think it's better to use the alternatives as stated in the
following pages:

https://web.archive.org/web/20221113170255/https://nix.dev/anti-patterns/language
https://toraritte.github.io/posts/2020-08-15-with-considered-harmful.html
https://github.com/NixOS/nix/issues/490

What do you think?

**Comment by vamega** [comment]
I've read those posts, and I think I also saw something from Tweag around Nickel and the complexity of dealing with `with`.

However I've not seen much if any usage of these patterns in any other parts of nixpkgs and chose to go with what was widely present in the repository instead.

If there's consensus that this is harmful, I imagine a treewide refactoring could take place.

In the end I'm indifferent, since I don't think I understand the subtleties enough to have a strong opinion at this time.

**Comment by ne9z** [comment]
My experience of writing flakes for my
computers is limited, but so far I have found a few advantages of the
`inherit` approach.

By explicitly using `inherit` instead of `with`, I can see where the
variables are coming from and control which variable is imported into
the scope, instead of importing the attrset wholesale.

This makes the nix file easier to reason with and helps with static
analysis, such as checking for unbounded variables without actually
evaluating the .

There are also some practical advantages.  For example, I previously had
to write

  myPkg = with pkgs; [ enchant nuspell hunspellDicts.en_US hunspellDicts.de_DE ];

with builtins.attrValues I can now write

  myPkg = builtins.attrValues {
    inherit (pkgs) enchant nuspell;
    inherit (pkgs.hunspellDicts) en_US de_DE;
  };


But I think keeping in the same style with other packages is also a
reasonable goal.  I would give approval for this one.

**Review by ghost — APPROVED** [review]

File: pkgs/applications/networking/p2p/pyrosimple/default.nix [hunk 1]
```diff
@@ -24,14 +26,27 @@
 }:
 
 let
-  pname = "pyrosimple";
-  version = "2.7.0";
+ pname = "pyrosimple";
+ version = "2.8.0";
+in buildPythonPackage {
+  inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:32]:
```suggestion
buildPythonPackage rec {
  pname = "pyrosimple";
  version = "2.8.0";
```

**Comment by ne9z** [comment]
As described in the links above, the `rec` construct is also an
anti-pattern, which I intentionally avoided while writing the
definition.

**Comment by SuperSandro2000** [comment]
> As described in the links above, the `rec` construct is also an anti-pattern, which I intentionally avoided while writing the definition.

The nix.dev link only applies to libraries and big expressions which is not the case. Using rec in mkDerivation is not persay an anti pattern.

**Pull request merged**